overlord/devicestate/handlers_install.go

```diff
@@ -315,10 +316,22 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 	// encryption is required in secured devices and optional in other grades
 	if err := secbootCheckKeySealingSupported(); err != nil {
 		if secured {
-			return false, fmt.Errorf("cannot encrypt secured device: %v", err)
+			return false, fmt.Errorf("cannot encrypt device storage as mandated by model grade secured: %v", err)
 		}
+		if encrypted {
+			return false, fmt.Errorf("cannot encrypt device storage as mandated by encrypted storage-safety model option: %v", err)
+		}
+		return false, nil
+	}
+
+	// TODO: provide way to select via install chooser menu
+	// if the install is unencrypted or encrypted
+	if model.StorageSafety() == asserts.StorageSafetyPreferUnencrypted {
+		logger.Noticef(`installing system unencrypted to comply with prefer-unencrypted storage-safety model option`)
```

Comment:
Would it make sense to have something like an install.json file in data/boot/save that would describe how the system was installed (eg. with data being (un-)encrypted due to one of force-unencrypted, bad TPM, model preference and so on?)

Reply:
It's probably mostly a @pedronis question but it maybe something useful, especially when hooks can also be the reason why we use encryption later.

Reply:
it might make sense, but is probably something to think about/design in the context of the factory process work, we shouldn't try to add it quickly now